addons/product/models/product_pricelist.py

```diff
@@ -96,7 +96,17 @@ def _compute_price_rule_get_items(self, products_qty_partner, date, uom_id, prod
         # Load all rules
         self.env['product.pricelist.item'].flush(['price', 'currency_id', 'company_id'])
         self.env.cr.execute(
-            """
+            self._query_price_rule_get_items(),
```

Comment:
what if query is changed by inheritance?

we should be able to modify params as well don't you think?

what about implement 
get_select
get_from
get_where 

somethin like 
https://github.com/odoo/odoo/blob/15.0/addons/purchase/report/purchase_report.py#L64

Reply:
I considered that, but I'm trying to leave changes as simple as possible, as we're in stable. If someone needs to add params, the method `cr.mogrify()` could be used to insert them.

Reply:
I think same than @hugho-ad 

But you have a good point

@mart-e What do you think?

Reply:
I'd like to add, even if I split the query into clauses, that won't ease adding params

Reply:
You are right!

Because it is using params args instead of keywords-args

I mean, it is using

```python
env.cr.execute("SELECT fields FROM table WHERE field1=%s", ('hello',))
```


Instead of 

```python
env.cr.execute("SELECT fields FROM table WHERE field1=%(value1)s", {'value1': 'hello'})
```

Using keywords-args it is easier to add more or edit the current values

But I agree it could be more changes but I even not sure If odoo is agree to merge this light version